cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -354,6 +356,10 @@ public TwillPreparer withConfiguration(Map<String, String> config) {
       workloadIdentityServiceAccount = config
         .get(KubeTwillRunnerService.WORKLOAD_IDENTITY_GCP_SERVICE_ACCOUNT_EMAIL_PROPERTY);
     }
+    if (config.containsKey(KubeTwillRunnerService.JOB_SKIP_CLEANUP)) {
+      kubeTwillConfigs.put(KubeTwillRunnerService.JOB_SKIP_CLEANUP,
```

Comment:
in general, for internal logic like this we should not be trying to future-proof, since we can change anything whenever we want. If it is needed in the future we can add it, otherwise we should be doing the simplest/clearest implementation.